mods/page_revisions.js

```diff
@@ -253,8 +253,9 @@ PRS.prototype.getTitleRevision = function(restbase, req) {
         throw new Error("Invalid revision: " + rp.revision);
     }
     return revisionRequest
-    .then(self._checkRevReturn)
     .then(function(res) {
+        // check if the revision has any restrictions
+        self._checkRevReturn(res);
```

Comment:
Yep, but we can't escape that since:
- we have to wait for the return from storage in order to check the restrictions
- there are no other `.then()` blocks where we could stick it